// @elasticemail/elasticemail-client/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AccessLevel", {
  enumerable: true,
  get: function get() {
    return _AccessLevel["default"];
  }
});
Object.defineProperty(exports, "AccountStatusEnum", {
  enumerable: true,
  get: function get() {
    return _AccountStatusEnum["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ApiKey", {
  enumerable: true,
  get: function get() {
    return _ApiKey["default"];
  }
});
Object.defineProperty(exports, "ApiKeyPayload", {
  enumerable: true,
  get: function get() {
    return _ApiKeyPayload["default"];
  }
});
Object.defineProperty(exports, "BodyContentType", {
  enumerable: true,
  get: function get() {
    return _BodyContentType["default"];
  }
});
Object.defineProperty(exports, "BodyPart", {
  enumerable: true,
  get: function get() {
    return _BodyPart["default"];
  }
});
Object.defineProperty(exports, "Campaign", {
  enumerable: true,
  get: function get() {
    return _Campaign["default"];
  }
});
Object.defineProperty(exports, "CampaignOptions", {
  enumerable: true,
  get: function get() {
    return _CampaignOptions["default"];
  }
});
Object.defineProperty(exports, "CampaignRecipient", {
  enumerable: true,
  get: function get() {
    return _CampaignRecipient["default"];
  }
});
Object.defineProperty(exports, "CampaignStatus", {
  enumerable: true,
  get: function get() {
    return _CampaignStatus["default"];
  }
});
Object.defineProperty(exports, "CampaignTemplate", {
  enumerable: true,
  get: function get() {
    return _CampaignTemplate["default"];
  }
});
Object.defineProperty(exports, "CampaignsApi", {
  enumerable: true,
  get: function get() {
    return _CampaignsApi["default"];
  }
});
Object.defineProperty(exports, "ChannelLogStatusSummary", {
  enumerable: true,
  get: function get() {
    return _ChannelLogStatusSummary["default"];
  }
});
Object.defineProperty(exports, "CompressionFormat", {
  enumerable: true,
  get: function get() {
    return _CompressionFormat["default"];
  }
});
Object.defineProperty(exports, "ConsentData", {
  enumerable: true,
  get: function get() {
    return _ConsentData["default"];
  }
});
Object.defineProperty(exports, "ConsentTracking", {
  enumerable: true,
  get: function get() {
    return _ConsentTracking["default"];
  }
});
Object.defineProperty(exports, "Contact", {
  enumerable: true,
  get: function get() {
    return _Contact["default"];
  }
});
Object.defineProperty(exports, "ContactActivity", {
  enumerable: true,
  get: function get() {
    return _ContactActivity["default"];
  }
});
Object.defineProperty(exports, "ContactPayload", {
  enumerable: true,
  get: function get() {
    return _ContactPayload["default"];
  }
});
Object.defineProperty(exports, "ContactSource", {
  enumerable: true,
  get: function get() {
    return _ContactSource["default"];
  }
});
Object.defineProperty(exports, "ContactStatus", {
  enumerable: true,
  get: function get() {
    return _ContactStatus["default"];
  }
});
Object.defineProperty(exports, "ContactUpdatePayload", {
  enumerable: true,
  get: function get() {
    return _ContactUpdatePayload["default"];
  }
});
Object.defineProperty(exports, "ContactsApi", {
  enumerable: true,
  get: function get() {
    return _ContactsApi["default"];
  }
});
Object.defineProperty(exports, "ContactsList", {
  enumerable: true,
  get: function get() {
    return _ContactsList["default"];
  }
});
Object.defineProperty(exports, "DeliveryOptimizationType", {
  enumerable: true,
  get: function get() {
    return _DeliveryOptimizationType["default"];
  }
});
Object.defineProperty(exports, "EmailContent", {
  enumerable: true,
  get: function get() {
    return _EmailContent["default"];
  }
});
Object.defineProperty(exports, "EmailData", {
  enumerable: true,
  get: function get() {
    return _EmailData["default"];
  }
});
Object.defineProperty(exports, "EmailMessageData", {
  enumerable: true,
  get: function get() {
    return _EmailMessageData["default"];
  }
});
Object.defineProperty(exports, "EmailRecipient", {
  enumerable: true,
  get: function get() {
    return _EmailRecipient["default"];
  }
});
Object.defineProperty(exports, "EmailSend", {
  enumerable: true,
  get: function get() {
    return _EmailSend["default"];
  }
});
Object.defineProperty(exports, "EmailStatus", {
  enumerable: true,
  get: function get() {
    return _EmailStatus["default"];
  }
});
Object.defineProperty(exports, "EmailTransactionalMessageData", {
  enumerable: true,
  get: function get() {
    return _EmailTransactionalMessageData["default"];
  }
});
Object.defineProperty(exports, "EmailValidationResult", {
  enumerable: true,
  get: function get() {
    return _EmailValidationResult["default"];
  }
});
Object.defineProperty(exports, "EmailValidationStatus", {
  enumerable: true,
  get: function get() {
    return _EmailValidationStatus["default"];
  }
});
Object.defineProperty(exports, "EmailView", {
  enumerable: true,
  get: function get() {
    return _EmailView["default"];
  }
});
Object.defineProperty(exports, "EmailsApi", {
  enumerable: true,
  get: function get() {
    return _EmailsApi["default"];
  }
});
Object.defineProperty(exports, "EmailsPayload", {
  enumerable: true,
  get: function get() {
    return _EmailsPayload["default"];
  }
});
Object.defineProperty(exports, "EncodingType", {
  enumerable: true,
  get: function get() {
    return _EncodingType["default"];
  }
});
Object.defineProperty(exports, "EventType", {
  enumerable: true,
  get: function get() {
    return _EventType["default"];
  }
});
Object.defineProperty(exports, "EventsApi", {
  enumerable: true,
  get: function get() {
    return _EventsApi["default"];
  }
});
Object.defineProperty(exports, "EventsOrderBy", {
  enumerable: true,
  get: function get() {
    return _EventsOrderBy["default"];
  }
});
Object.defineProperty(exports, "ExportFileFormats", {
  enumerable: true,
  get: function get() {
    return _ExportFileFormats["default"];
  }
});
Object.defineProperty(exports, "ExportLink", {
  enumerable: true,
  get: function get() {
    return _ExportLink["default"];
  }
});
Object.defineProperty(exports, "ExportStatus", {
  enumerable: true,
  get: function get() {
    return _ExportStatus["default"];
  }
});
Object.defineProperty(exports, "FileInfo", {
  enumerable: true,
  get: function get() {
    return _FileInfo["default"];
  }
});
Object.defineProperty(exports, "FilePayload", {
  enumerable: true,
  get: function get() {
    return _FilePayload["default"];
  }
});
Object.defineProperty(exports, "FileUploadResult", {
  enumerable: true,
  get: function get() {
    return _FileUploadResult["default"];
  }
});
Object.defineProperty(exports, "FilesApi", {
  enumerable: true,
  get: function get() {
    return _FilesApi["default"];
  }
});
Object.defineProperty(exports, "InboundPayload", {
  enumerable: true,
  get: function get() {
    return _InboundPayload["default"];
  }
});
Object.defineProperty(exports, "InboundRoute", {
  enumerable: true,
  get: function get() {
    return _InboundRoute["default"];
  }
});
Object.defineProperty(exports, "InboundRouteActionType", {
  enumerable: true,
  get: function get() {
    return _InboundRouteActionType["default"];
  }
});
Object.defineProperty(exports, "InboundRouteApi", {
  enumerable: true,
  get: function get() {
    return _InboundRouteApi["default"];
  }
});
Object.defineProperty(exports, "InboundRouteFilterType", {
  enumerable: true,
  get: function get() {
    return _InboundRouteFilterType["default"];
  }
});
Object.defineProperty(exports, "ListPayload", {
  enumerable: true,
  get: function get() {
    return _ListPayload["default"];
  }
});
Object.defineProperty(exports, "ListUpdatePayload", {
  enumerable: true,
  get: function get() {
    return _ListUpdatePayload["default"];
  }
});
Object.defineProperty(exports, "ListsApi", {
  enumerable: true,
  get: function get() {
    return _ListsApi["default"];
  }
});
Object.defineProperty(exports, "LogJobStatus", {
  enumerable: true,
  get: function get() {
    return _LogJobStatus["default"];
  }
});
Object.defineProperty(exports, "LogStatusSummary", {
  enumerable: true,
  get: function get() {
    return _LogStatusSummary["default"];
  }
});
Object.defineProperty(exports, "MergeEmailPayload", {
  enumerable: true,
  get: function get() {
    return _MergeEmailPayload["default"];
  }
});
Object.defineProperty(exports, "MessageAttachment", {
  enumerable: true,
  get: function get() {
    return _MessageAttachment["default"];
  }
});
Object.defineProperty(exports, "MessageCategory", {
  enumerable: true,
  get: function get() {
    return _MessageCategory["default"];
  }
});
Object.defineProperty(exports, "NewApiKey", {
  enumerable: true,
  get: function get() {
    return _NewApiKey["default"];
  }
});
Object.defineProperty(exports, "NewSmtpCredentials", {
  enumerable: true,
  get: function get() {
    return _NewSmtpCredentials["default"];
  }
});
Object.defineProperty(exports, "Options", {
  enumerable: true,
  get: function get() {
    return _Options["default"];
  }
});
Object.defineProperty(exports, "RecipientEvent", {
  enumerable: true,
  get: function get() {
    return _RecipientEvent["default"];
  }
});
Object.defineProperty(exports, "SecurityApi", {
  enumerable: true,
  get: function get() {
    return _SecurityApi["default"];
  }
});
Object.defineProperty(exports, "Segment", {
  enumerable: true,
  get: function get() {
    return _Segment["default"];
  }
});
Object.defineProperty(exports, "SegmentPayload", {
  enumerable: true,
  get: function get() {
    return _SegmentPayload["default"];
  }
});
Object.defineProperty(exports, "SegmentsApi", {
  enumerable: true,
  get: function get() {
    return _SegmentsApi["default"];
  }
});
Object.defineProperty(exports, "SmtpCredentials", {
  enumerable: true,
  get: function get() {
    return _SmtpCredentials["default"];
  }
});
Object.defineProperty(exports, "SmtpCredentialsPayload", {
  enumerable: true,
  get: function get() {
    return _SmtpCredentialsPayload["default"];
  }
});
Object.defineProperty(exports, "SortOrderItem", {
  enumerable: true,
  get: function get() {
    return _SortOrderItem["default"];
  }
});
Object.defineProperty(exports, "SplitOptimizationType", {
  enumerable: true,
  get: function get() {
    return _SplitOptimizationType["default"];
  }
});
Object.defineProperty(exports, "SplitOptions", {
  enumerable: true,
  get: function get() {
    return _SplitOptions["default"];
  }
});
Object.defineProperty(exports, "StatisticsApi", {
  enumerable: true,
  get: function get() {
    return _StatisticsApi["default"];
  }
});
Object.defineProperty(exports, "SubAccountInfo", {
  enumerable: true,
  get: function get() {
    return _SubAccountInfo["default"];
  }
});
Object.defineProperty(exports, "SubAccountsApi", {
  enumerable: true,
  get: function get() {
    return _SubAccountsApi["default"];
  }
});
Object.defineProperty(exports, "SubaccountEmailCreditsPayload", {
  enumerable: true,
  get: function get() {
    return _SubaccountEmailCreditsPayload["default"];
  }
});
Object.defineProperty(exports, "SubaccountEmailSettings", {
  enumerable: true,
  get: function get() {
    return _SubaccountEmailSettings["default"];
  }
});
Object.defineProperty(exports, "SubaccountEmailSettingsPayload", {
  enumerable: true,
  get: function get() {
    return _SubaccountEmailSettingsPayload["default"];
  }
});
Object.defineProperty(exports, "SubaccountPayload", {
  enumerable: true,
  get: function get() {
    return _SubaccountPayload["default"];
  }
});
Object.defineProperty(exports, "SubaccountSettingsInfo", {
  enumerable: true,
  get: function get() {
    return _SubaccountSettingsInfo["default"];
  }
});
Object.defineProperty(exports, "SubaccountSettingsInfoPayload", {
  enumerable: true,
  get: function get() {
    return _SubaccountSettingsInfoPayload["default"];
  }
});
Object.defineProperty(exports, "Suppression", {
  enumerable: true,
  get: function get() {
    return _Suppression["default"];
  }
});
Object.defineProperty(exports, "SuppressionsApi", {
  enumerable: true,
  get: function get() {
    return _SuppressionsApi["default"];
  }
});
Object.defineProperty(exports, "Template", {
  enumerable: true,
  get: function get() {
    return _Template["default"];
  }
});
Object.defineProperty(exports, "TemplatePayload", {
  enumerable: true,
  get: function get() {
    return _TemplatePayload["default"];
  }
});
Object.defineProperty(exports, "TemplateScope", {
  enumerable: true,
  get: function get() {
    return _TemplateScope["default"];
  }
});
Object.defineProperty(exports, "TemplateType", {
  enumerable: true,
  get: function get() {
    return _TemplateType["default"];
  }
});
Object.defineProperty(exports, "TemplatesApi", {
  enumerable: true,
  get: function get() {
    return _TemplatesApi["default"];
  }
});
Object.defineProperty(exports, "TransactionalRecipient", {
  enumerable: true,
  get: function get() {
    return _TransactionalRecipient["default"];
  }
});
Object.defineProperty(exports, "Utm", {
  enumerable: true,
  get: function get() {
    return _Utm["default"];
  }
});
Object.defineProperty(exports, "VerificationFileResult", {
  enumerable: true,
  get: function get() {
    return _VerificationFileResult["default"];
  }
});
Object.defineProperty(exports, "VerificationFileResultDetails", {
  enumerable: true,
  get: function get() {
    return _VerificationFileResultDetails["default"];
  }
});
Object.defineProperty(exports, "VerificationStatus", {
  enumerable: true,
  get: function get() {
    return _VerificationStatus["default"];
  }
});
Object.defineProperty(exports, "VerificationsApi", {
  enumerable: true,
  get: function get() {
    return _VerificationsApi["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _AccessLevel = _interopRequireDefault(require("./model/AccessLevel"));
var _AccountStatusEnum = _interopRequireDefault(require("./model/AccountStatusEnum"));
var _ApiKey = _interopRequireDefault(require("./model/ApiKey"));
var _ApiKeyPayload = _interopRequireDefault(require("./model/ApiKeyPayload"));
var _BodyContentType = _interopRequireDefault(require("./model/BodyContentType"));
var _BodyPart = _interopRequireDefault(require("./model/BodyPart"));
var _Campaign = _interopRequireDefault(require("./model/Campaign"));
var _CampaignOptions = _interopRequireDefault(require("./model/CampaignOptions"));
var _CampaignRecipient = _interopRequireDefault(require("./model/CampaignRecipient"));
var _CampaignStatus = _interopRequireDefault(require("./model/CampaignStatus"));
var _CampaignTemplate = _interopRequireDefault(require("./model/CampaignTemplate"));
var _ChannelLogStatusSummary = _interopRequireDefault(require("./model/ChannelLogStatusSummary"));
var _CompressionFormat = _interopRequireDefault(require("./model/CompressionFormat"));
var _ConsentData = _interopRequireDefault(require("./model/ConsentData"));
var _ConsentTracking = _interopRequireDefault(require("./model/ConsentTracking"));
var _Contact = _interopRequireDefault(require("./model/Contact"));
var _ContactActivity = _interopRequireDefault(require("./model/ContactActivity"));
var _ContactPayload = _interopRequireDefault(require("./model/ContactPayload"));
var _ContactSource = _interopRequireDefault(require("./model/ContactSource"));
var _ContactStatus = _interopRequireDefault(require("./model/ContactStatus"));
var _ContactUpdatePayload = _interopRequireDefault(require("./model/ContactUpdatePayload"));
var _ContactsList = _interopRequireDefault(require("./model/ContactsList"));
var _DeliveryOptimizationType = _interopRequireDefault(require("./model/DeliveryOptimizationType"));
var _EmailContent = _interopRequireDefault(require("./model/EmailContent"));
var _EmailData = _interopRequireDefault(require("./model/EmailData"));
var _EmailMessageData = _interopRequireDefault(require("./model/EmailMessageData"));
var _EmailRecipient = _interopRequireDefault(require("./model/EmailRecipient"));
var _EmailSend = _interopRequireDefault(require("./model/EmailSend"));
var _EmailStatus = _interopRequireDefault(require("./model/EmailStatus"));
var _EmailTransactionalMessageData = _interopRequireDefault(require("./model/EmailTransactionalMessageData"));
var _EmailValidationResult = _interopRequireDefault(require("./model/EmailValidationResult"));
var _EmailValidationStatus = _interopRequireDefault(require("./model/EmailValidationStatus"));
var _EmailView = _interopRequireDefault(require("./model/EmailView"));
var _EmailsPayload = _interopRequireDefault(require("./model/EmailsPayload"));
var _EncodingType = _interopRequireDefault(require("./model/EncodingType"));
var _EventType = _interopRequireDefault(require("./model/EventType"));
var _EventsOrderBy = _interopRequireDefault(require("./model/EventsOrderBy"));
var _ExportFileFormats = _interopRequireDefault(require("./model/ExportFileFormats"));
var _ExportLink = _interopRequireDefault(require("./model/ExportLink"));
var _ExportStatus = _interopRequireDefault(require("./model/ExportStatus"));
var _FileInfo = _interopRequireDefault(require("./model/FileInfo"));
var _FilePayload = _interopRequireDefault(require("./model/FilePayload"));
var _FileUploadResult = _interopRequireDefault(require("./model/FileUploadResult"));
var _InboundPayload = _interopRequireDefault(require("./model/InboundPayload"));
var _InboundRoute = _interopRequireDefault(require("./model/InboundRoute"));
var _InboundRouteActionType = _interopRequireDefault(require("./model/InboundRouteActionType"));
var _InboundRouteFilterType = _interopRequireDefault(require("./model/InboundRouteFilterType"));
var _ListPayload = _interopRequireDefault(require("./model/ListPayload"));
var _ListUpdatePayload = _interopRequireDefault(require("./model/ListUpdatePayload"));
var _LogJobStatus = _interopRequireDefault(require("./model/LogJobStatus"));
var _LogStatusSummary = _interopRequireDefault(require("./model/LogStatusSummary"));
var _MergeEmailPayload = _interopRequireDefault(require("./model/MergeEmailPayload"));
var _MessageAttachment = _interopRequireDefault(require("./model/MessageAttachment"));
var _MessageCategory = _interopRequireDefault(require("./model/MessageCategory"));
var _NewApiKey = _interopRequireDefault(require("./model/NewApiKey"));
var _NewSmtpCredentials = _interopRequireDefault(require("./model/NewSmtpCredentials"));
var _Options = _interopRequireDefault(require("./model/Options"));
var _RecipientEvent = _interopRequireDefault(require("./model/RecipientEvent"));
var _Segment = _interopRequireDefault(require("./model/Segment"));
var _SegmentPayload = _interopRequireDefault(require("./model/SegmentPayload"));
var _SmtpCredentials = _interopRequireDefault(require("./model/SmtpCredentials"));
var _SmtpCredentialsPayload = _interopRequireDefault(require("./model/SmtpCredentialsPayload"));
var _SortOrderItem = _interopRequireDefault(require("./model/SortOrderItem"));
var _SplitOptimizationType = _interopRequireDefault(require("./model/SplitOptimizationType"));
var _SplitOptions = _interopRequireDefault(require("./model/SplitOptions"));
var _SubAccountInfo = _interopRequireDefault(require("./model/SubAccountInfo"));
var _SubaccountEmailCreditsPayload = _interopRequireDefault(require("./model/SubaccountEmailCreditsPayload"));
var _SubaccountEmailSettings = _interopRequireDefault(require("./model/SubaccountEmailSettings"));
var _SubaccountEmailSettingsPayload = _interopRequireDefault(require("./model/SubaccountEmailSettingsPayload"));
var _SubaccountPayload = _interopRequireDefault(require("./model/SubaccountPayload"));
var _SubaccountSettingsInfo = _interopRequireDefault(require("./model/SubaccountSettingsInfo"));
var _SubaccountSettingsInfoPayload = _interopRequireDefault(require("./model/SubaccountSettingsInfoPayload"));
var _Suppression = _interopRequireDefault(require("./model/Suppression"));
var _Template = _interopRequireDefault(require("./model/Template"));
var _TemplatePayload = _interopRequireDefault(require("./model/TemplatePayload"));
var _TemplateScope = _interopRequireDefault(require("./model/TemplateScope"));
var _TemplateType = _interopRequireDefault(require("./model/TemplateType"));
var _TransactionalRecipient = _interopRequireDefault(require("./model/TransactionalRecipient"));
var _Utm = _interopRequireDefault(require("./model/Utm"));
var _VerificationFileResult = _interopRequireDefault(require("./model/VerificationFileResult"));
var _VerificationFileResultDetails = _interopRequireDefault(require("./model/VerificationFileResultDetails"));
var _VerificationStatus = _interopRequireDefault(require("./model/VerificationStatus"));
var _CampaignsApi = _interopRequireDefault(require("./api/CampaignsApi"));
var _ContactsApi = _interopRequireDefault(require("./api/ContactsApi"));
var _EmailsApi = _interopRequireDefault(require("./api/EmailsApi"));
var _EventsApi = _interopRequireDefault(require("./api/EventsApi"));
var _FilesApi = _interopRequireDefault(require("./api/FilesApi"));
var _InboundRouteApi = _interopRequireDefault(require("./api/InboundRouteApi"));
var _ListsApi = _interopRequireDefault(require("./api/ListsApi"));
var _SecurityApi = _interopRequireDefault(require("./api/SecurityApi"));
var _SegmentsApi = _interopRequireDefault(require("./api/SegmentsApi"));
var _StatisticsApi = _interopRequireDefault(require("./api/StatisticsApi"));
var _SubAccountsApi = _interopRequireDefault(require("./api/SubAccountsApi"));
var _SuppressionsApi = _interopRequireDefault(require("./api/SuppressionsApi"));
var _TemplatesApi = _interopRequireDefault(require("./api/TemplatesApi"));
var _VerificationsApi = _interopRequireDefault(require("./api/VerificationsApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
