// @elasticemail/elasticemail-client/dist/model/EmailValidationResult.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailValidationStatus = _interopRequireDefault(require("./EmailValidationStatus"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Elastic Email REST API
                                                                                                                                                                                                                                                                                                                                                                                               * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@elasticemail.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The EmailValidationResult model module.
 * @module model/EmailValidationResult
 * @version 4.0.23
 */
var EmailValidationResult = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EmailValidationResult</code>.
   * @alias module:model/EmailValidationResult
   */
  function EmailValidationResult() {
    _classCallCheck(this, EmailValidationResult);
    EmailValidationResult.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(EmailValidationResult, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>EmailValidationResult</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EmailValidationResult} obj Optional instance to populate.
     * @return {module:model/EmailValidationResult} The populated <code>EmailValidationResult</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailValidationResult();
        if (data.hasOwnProperty('Account')) {
          obj['Account'] = _ApiClient["default"].convertToType(data['Account'], 'String');
        }
        if (data.hasOwnProperty('Domain')) {
          obj['Domain'] = _ApiClient["default"].convertToType(data['Domain'], 'String');
        }
        if (data.hasOwnProperty('Email')) {
          obj['Email'] = _ApiClient["default"].convertToType(data['Email'], 'String');
        }
        if (data.hasOwnProperty('SuggestedSpelling')) {
          obj['SuggestedSpelling'] = _ApiClient["default"].convertToType(data['SuggestedSpelling'], 'String');
        }
        if (data.hasOwnProperty('Disposable')) {
          obj['Disposable'] = _ApiClient["default"].convertToType(data['Disposable'], 'Boolean');
        }
        if (data.hasOwnProperty('Role')) {
          obj['Role'] = _ApiClient["default"].convertToType(data['Role'], 'Boolean');
        }
        if (data.hasOwnProperty('Reason')) {
          obj['Reason'] = _ApiClient["default"].convertToType(data['Reason'], 'String');
        }
        if (data.hasOwnProperty('DateAdded')) {
          obj['DateAdded'] = _ApiClient["default"].convertToType(data['DateAdded'], 'Date');
        }
        if (data.hasOwnProperty('Result')) {
          obj['Result'] = _EmailValidationStatus["default"].constructFromObject(data['Result']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>EmailValidationResult</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EmailValidationResult</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Account'] && !(typeof data['Account'] === 'string' || data['Account'] instanceof String)) {
        throw new Error("Expected the field `Account` to be a primitive type in the JSON string but got " + data['Account']);
      }
      // ensure the json data is a string
      if (data['Domain'] && !(typeof data['Domain'] === 'string' || data['Domain'] instanceof String)) {
        throw new Error("Expected the field `Domain` to be a primitive type in the JSON string but got " + data['Domain']);
      }
      // ensure the json data is a string
      if (data['Email'] && !(typeof data['Email'] === 'string' || data['Email'] instanceof String)) {
        throw new Error("Expected the field `Email` to be a primitive type in the JSON string but got " + data['Email']);
      }
      // ensure the json data is a string
      if (data['SuggestedSpelling'] && !(typeof data['SuggestedSpelling'] === 'string' || data['SuggestedSpelling'] instanceof String)) {
        throw new Error("Expected the field `SuggestedSpelling` to be a primitive type in the JSON string but got " + data['SuggestedSpelling']);
      }
      // ensure the json data is a string
      if (data['Reason'] && !(typeof data['Reason'] === 'string' || data['Reason'] instanceof String)) {
        throw new Error("Expected the field `Reason` to be a primitive type in the JSON string but got " + data['Reason']);
      }
      return true;
    }
  }]);
  return EmailValidationResult;
}();
/**
 * Local part of an email
 * @member {String} Account
 */
EmailValidationResult.prototype['Account'] = undefined;

/**
 * Name of selected domain.
 * @member {String} Domain
 */
EmailValidationResult.prototype['Domain'] = undefined;

/**
 * Full email address that was verified
 * @member {String} Email
 */
EmailValidationResult.prototype['Email'] = undefined;

/**
 * Suggested spelling if a possible mistake was found
 * @member {String} SuggestedSpelling
 */
EmailValidationResult.prototype['SuggestedSpelling'] = undefined;

/**
 * Does the email have a temporary domain
 * @member {Boolean} Disposable
 */
EmailValidationResult.prototype['Disposable'] = undefined;

/**
 * Is an email a role email (e.g. info@, noreply@ etc.)
 * @member {Boolean} Role
 */
EmailValidationResult.prototype['Role'] = undefined;

/**
 * All detected issues
 * @member {String} Reason
 */
EmailValidationResult.prototype['Reason'] = undefined;

/**
 * Date of creation in YYYY-MM-DDThh:ii:ss format
 * @member {Date} DateAdded
 */
EmailValidationResult.prototype['DateAdded'] = undefined;

/**
 * @member {module:model/EmailValidationStatus} Result
 */
EmailValidationResult.prototype['Result'] = undefined;
var _default = EmailValidationResult;
exports["default"] = _default;
