// @elasticemail/elasticemail-client/dist/model/NewSmtpCredentials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AccessLevel = _interopRequireDefault(require("./AccessLevel"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Elastic Email REST API
                                                                                                                                                                                                                                                                                                                                                                                               * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@elasticemail.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The NewSmtpCredentials model module.
 * @module model/NewSmtpCredentials
 * @version 4.0.23
 */
var NewSmtpCredentials = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>NewSmtpCredentials</code>.
   * Newly generated SMTP Credentials with Token
   * @alias module:model/NewSmtpCredentials
   */
  function NewSmtpCredentials() {
    _classCallCheck(this, NewSmtpCredentials);
    NewSmtpCredentials.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(NewSmtpCredentials, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>NewSmtpCredentials</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/NewSmtpCredentials} obj Optional instance to populate.
     * @return {module:model/NewSmtpCredentials} The populated <code>NewSmtpCredentials</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new NewSmtpCredentials();
        if (data.hasOwnProperty('Token')) {
          obj['Token'] = _ApiClient["default"].convertToType(data['Token'], 'String');
        }
        if (data.hasOwnProperty('AccessLevel')) {
          obj['AccessLevel'] = _AccessLevel["default"].constructFromObject(data['AccessLevel']);
        }
        if (data.hasOwnProperty('Name')) {
          obj['Name'] = _ApiClient["default"].convertToType(data['Name'], 'String');
        }
        if (data.hasOwnProperty('DateCreated')) {
          obj['DateCreated'] = _ApiClient["default"].convertToType(data['DateCreated'], 'Date');
        }
        if (data.hasOwnProperty('LastUse')) {
          obj['LastUse'] = _ApiClient["default"].convertToType(data['LastUse'], 'Date');
        }
        if (data.hasOwnProperty('Expires')) {
          obj['Expires'] = _ApiClient["default"].convertToType(data['Expires'], 'Date');
        }
        if (data.hasOwnProperty('RestrictAccessToIPRange')) {
          obj['RestrictAccessToIPRange'] = _ApiClient["default"].convertToType(data['RestrictAccessToIPRange'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>NewSmtpCredentials</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>NewSmtpCredentials</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Token'] && !(typeof data['Token'] === 'string' || data['Token'] instanceof String)) {
        throw new Error("Expected the field `Token` to be a primitive type in the JSON string but got " + data['Token']);
      }
      // ensure the json data is a string
      if (data['Name'] && !(typeof data['Name'] === 'string' || data['Name'] instanceof String)) {
        throw new Error("Expected the field `Name` to be a primitive type in the JSON string but got " + data['Name']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['RestrictAccessToIPRange'])) {
        throw new Error("Expected the field `RestrictAccessToIPRange` to be an array in the JSON data but got " + data['RestrictAccessToIPRange']);
      }
      return true;
    }
  }]);
  return NewSmtpCredentials;
}();
/**
 * Unique token to be used in the system
 * @member {String} Token
 */
NewSmtpCredentials.prototype['Token'] = undefined;

/**
 * @member {module:model/AccessLevel} AccessLevel
 */
NewSmtpCredentials.prototype['AccessLevel'] = undefined;

/**
 * Name of the key.
 * @member {String} Name
 */
NewSmtpCredentials.prototype['Name'] = undefined;

/**
 * Date this SmtpCredential was created.
 * @member {Date} DateCreated
 */
NewSmtpCredentials.prototype['DateCreated'] = undefined;

/**
 * Date this SmtpCredential was last used.
 * @member {Date} LastUse
 */
NewSmtpCredentials.prototype['LastUse'] = undefined;

/**
 * Date this SmtpCredential expires.
 * @member {Date} Expires
 */
NewSmtpCredentials.prototype['Expires'] = undefined;

/**
 * Which IPs can use this SmtpCredential
 * @member {Array.<String>} RestrictAccessToIPRange
 */
NewSmtpCredentials.prototype['RestrictAccessToIPRange'] = undefined;
var _default = NewSmtpCredentials;
exports["default"] = _default;
