// @elasticemail/elasticemail-client/dist/model/VerificationFileResultDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailValidationResult = _interopRequireDefault(require("./EmailValidationResult"));
var _FileUploadResult = _interopRequireDefault(require("./FileUploadResult"));
var _VerificationStatus = _interopRequireDefault(require("./VerificationStatus"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Elastic Email REST API
                                                                                                                                                                                                                                                                                                                                                                                               * This API is based on the REST API architecture, allowing the user to easily manage their data with this resource-based approach.    Every API call is established on which specific request type (GET, POST, PUT, DELETE) will be used.    The API has a limit of 20 concurrent connections and a hard timeout of 600 seconds per request.    To start using this API, you will need your Access Token (available <a target=\"_blank\" href=\"https://app.elasticemail.com/marketing/settings/new/manage-api\">here</a>). Remember to keep it safe. Required access levels are listed in the given request’s description.    Downloadable library clients can be found in our Github repository <a target=\"_blank\" href=\"https://github.com/ElasticEmail?tab=repositories&q=%22rest+api%22+in%3Areadme\">here</a>
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 4.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@elasticemail.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The VerificationFileResultDetails model module.
 * @module model/VerificationFileResultDetails
 * @version 4.0.23
 */
var VerificationFileResultDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerificationFileResultDetails</code>.
   * Detailed verification file result info
   * @alias module:model/VerificationFileResultDetails
   */
  function VerificationFileResultDetails() {
    _classCallCheck(this, VerificationFileResultDetails);
    VerificationFileResultDetails.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(VerificationFileResultDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>VerificationFileResultDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerificationFileResultDetails} obj Optional instance to populate.
     * @return {module:model/VerificationFileResultDetails} The populated <code>VerificationFileResultDetails</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerificationFileResultDetails();
        if (data.hasOwnProperty('VerificationResult')) {
          obj['VerificationResult'] = _ApiClient["default"].convertToType(data['VerificationResult'], [_EmailValidationResult["default"]]);
        }
        if (data.hasOwnProperty('VerificationID')) {
          obj['VerificationID'] = _ApiClient["default"].convertToType(data['VerificationID'], 'String');
        }
        if (data.hasOwnProperty('Filename')) {
          obj['Filename'] = _ApiClient["default"].convertToType(data['Filename'], 'String');
        }
        if (data.hasOwnProperty('VerificationStatus')) {
          obj['VerificationStatus'] = _VerificationStatus["default"].constructFromObject(data['VerificationStatus']);
        }
        if (data.hasOwnProperty('FileUploadResult')) {
          obj['FileUploadResult'] = _FileUploadResult["default"].constructFromObject(data['FileUploadResult']);
        }
        if (data.hasOwnProperty('DateAdded')) {
          obj['DateAdded'] = _ApiClient["default"].convertToType(data['DateAdded'], 'Date');
        }
        if (data.hasOwnProperty('Source')) {
          obj['Source'] = _ApiClient["default"].convertToType(data['Source'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>VerificationFileResultDetails</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>VerificationFileResultDetails</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['VerificationResult']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['VerificationResult'])) {
          throw new Error("Expected the field `VerificationResult` to be an array in the JSON data but got " + data['VerificationResult']);
        }
        // validate the optional field `VerificationResult` (array)
        var _iterator = _createForOfIteratorHelper(data['VerificationResult']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _EmailValidationResult["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['VerificationID'] && !(typeof data['VerificationID'] === 'string' || data['VerificationID'] instanceof String)) {
        throw new Error("Expected the field `VerificationID` to be a primitive type in the JSON string but got " + data['VerificationID']);
      }
      // ensure the json data is a string
      if (data['Filename'] && !(typeof data['Filename'] === 'string' || data['Filename'] instanceof String)) {
        throw new Error("Expected the field `Filename` to be a primitive type in the JSON string but got " + data['Filename']);
      }
      // validate the optional field `FileUploadResult`
      if (data['FileUploadResult']) {
        // data not null
        _FileUploadResult["default"].validateJSON(data['FileUploadResult']);
      }
      // ensure the json data is a string
      if (data['Source'] && !(typeof data['Source'] === 'string' || data['Source'] instanceof String)) {
        throw new Error("Expected the field `Source` to be a primitive type in the JSON string but got " + data['Source']);
      }
      return true;
    }
  }]);
  return VerificationFileResultDetails;
}();
/**
 * Verification result's details
 * @member {Array.<module:model/EmailValidationResult>} VerificationResult
 */
VerificationFileResultDetails.prototype['VerificationResult'] = undefined;

/**
 * Identifier of this verification result
 * @member {String} VerificationID
 */
VerificationFileResultDetails.prototype['VerificationID'] = undefined;

/**
 * Origin file name
 * @member {String} Filename
 */
VerificationFileResultDetails.prototype['Filename'] = undefined;

/**
 * @member {module:model/VerificationStatus} VerificationStatus
 */
VerificationFileResultDetails.prototype['VerificationStatus'] = undefined;

/**
 * @member {module:model/FileUploadResult} FileUploadResult
 */
VerificationFileResultDetails.prototype['FileUploadResult'] = undefined;

/**
 * Date of creation in YYYY-MM-DDThh:ii:ss format
 * @member {Date} DateAdded
 */
VerificationFileResultDetails.prototype['DateAdded'] = undefined;

/**
 * Origin file extension
 * @member {String} Source
 */
VerificationFileResultDetails.prototype['Source'] = undefined;
var _default = VerificationFileResultDetails;
exports["default"] = _default;
